sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRegistryClientTests.cs

```diff
@@ -17,10 +18,15 @@ public ContainerRegistryClientTests(bool isAsync) : base(isAsync)
         private ContainerRegistryClient client { get; set; }
         private readonly Uri _url = new Uri("https://example.azurecr.io");
 
+        private TokenCredential GetCredential()
+        {
+            return new EnvironmentCredential();
```

Comment:
What about using [TestEnvironment.Credential](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs#L156)?

This is RecordedTestMode aware, so it will give you a MockCredential when in playback mode.

Reply:
These ones are unit tests inheriting from ClientTestBase, so I don't have access to that here.